src/util/simplify_expr.cpp

```diff
@@ -2617,19 +2617,35 @@ bool simplify_exprt::simplify_rec(exprt &expr)
   exprt tmp=expr;
   bool result=true;
 
+  if(!tmp.get_bool(ID_C_expr_simplified))
+  {
   result=simplify_node_preorder(tmp);
 
   if(!simplify_node(tmp)) result=false;
 
-  #if 1
   replace_mapt::const_iterator it=local_replace_map.find(tmp);
   if(it!=local_replace_map.end())
   {
     tmp=it->second;
     result=false;
   }
-  #else
-  if(!local_replace_map.empty() &&
+  }
+  #if 0
```

Comment:
@tautschnig, can you explain why this code is commented out?
Otherwise, this looks mergeable.